parser-typechecker/src/Unison/PatternMatchCoverage/Fix.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Unison.PatternMatchCoverage.Fix where
+
+newtype Fix f = Fix {unFix :: f (Fix f)}
```

Comment:
Just curious, any particular reason to add our own rather than use [`Data.Fix`](https://hackage.haskell.org/package/data-fix-0.3.2/docs/Data-Fix.html) (it's already in our transitive dependencies).

I can see us probably using more fix-points and recursion schemes in the future so I think I'd prefer not to have duplicate Fix types floating around if possible 😄

Reply:
I used it because it was easier to define some instances with quantified constraints instead of adding `Show1`, `Eq1`, and friends to all the base functors.

The use of `Fix` should be purely internal to the `PatternMatchCoverage` subtree; it isn't exposed in the (very small) interface given in `Unison.PatternMatchCoverage`.